src/datepicker/datepicker.js

```diff
@@ -219,7 +219,19 @@ export default class Datepicker extends React.Component<
         }
       }
     } else {
-      const date = new Date(inputValue);
+      const dateString = this.normalizeDashes(inputValue);
+      let date = new Date(dateString);
+      const formatString = this.props.formatString;
+      if (formatString) {
+        // Prevent early parsing of value.
+        // Eg 25.12.2 will be transformed to 25.12.0002 formatted from date to string
+        if (dateString.replace(/ /g, '').length < formatString.length) {
```

Comment:
What is the regex supposed to look for? Also, wouldn't whitespace be valid in some formatting cases?

Reply:
This regex is erasing whitespaces generated by masked input. But there is a good catch. I should actually compare replaced whitespaces for both of them.
Here is `mask="99 99 9999"` and `formatString="dd MM YYYY"`, if your input value is `"1"` then masked input will transform it to `"1         "` . 
So, we should compare `"1".length` to `"ddMMYYYY".length`

Reply:
Correct, you'd need to trim the `dateString` or  get rid of spaces in both strings that are being compared. 
Regarding the space regex, I'm thinking we should look for any whitespace `/s`.

Reply:
Done + added extra test